docker/server/entrypoint.sh

```
@@ -129,20 +158,94 @@ mirror_hexclave_stack_env
+  if ! _identity=$(cat /app/.stack-runtime-build-identity 2>/dev/null) || [ -z "$_identity" ]; then
+    return 1
+  fi
+  printf '%s\n' "$_identity"
+}
+rebuild_runtime_tree() {
+  if [ "$WORK_DIR" = "/app" ]; then
+    echo "ERROR: STACK_RUNTIME_WORK_DIR=/app cannot rebuild changed sentinel values in place. Recreate the container without STACK_RUNTIME_WORK_DIR=/app." >&2
+    exit 1
+  fi
+  while IFS= read -r -d '' _runtime_entry; do
+    rm -rf "$WORK_DIR/$_runtime_entry"
```

Shellcheck (0.11.0)

[warning] Use "${var:?}" to ensure this never expands to / .

(SC2115)